Apply feedback on build smart contracts page index (#436)
polkadot-developers/polkadot-docs: develop/smart-contracts/overview.md

---
title: Smart Contracts Overview

> description: Learn about smart contract development capabilities on the Polkadot ecosystem, either by leveraging the Asset Hub parachain or other alternatives.

Reviewer: ```suggestion
description: Learn about smart contract development capabilities in the Polkadot ecosystem, either by leveraging the Asset Hub parachain or other alternatives.
```

---

# Smart Contracts on Polkadot

## Native Smart Contracts

> ### Introduction

Reviewer: This order is sorta weird to me. Especially considering we don't have an intro section under "Other Smart Contract Environments"

Reviewer: Perhaps we should have an introduction for the page summarizing the entire page. Then go directly intoNative Smart Contracts and have the text directly under there (no Introduction header)

Author: Do you mean doing something like this?

```
# Smart Contracts on Polkadot

## Introduction
[general intro about the entire page here]

### Native Smart Contracts
### Other Smart Contract Environments
```

I do like that approach; the only thing we should consider in that case is how to transform the sub-titles of 'native smart contracts' and 'other smart contract environments' so they don't get too nested. Would you agree @eshaben ?

Reviewer: Native Smart Contracts and Other Smart Contracts can both be H2s (##)

Reviewer: So we should do something like this:

```
# Smart Contracts on Polkadot

## Introduction
[general intro about the entire page here]

## Native Smart Contracts

Asset Hub, Polkadot's system parachain, enables smart contract deployment and execution through PolkaVM, a cutting-edge virtual machine designed specifically for the Polkadot ecosystem. This native integration allows developers to deploy smart contracts directly on Polkadot's system chain while maintaining compatibility with Ethereum development tools and workflows.

### Smart Contract Development

...continue with rest of sections...

## Other Smart Contract Environments

...this looks good as you have it now...
```

Author: I tweaked the page a bit according to this. Here's the commit - [a95797c](https://github.com/polkadot-developers/polkadot-docs/pull/436/commits/a95797c4347e96a9c40aa6ba792a63671bc7991b)

# Smart Contracts on Polkadot

## Introduction

Polkadot offers developers multiple approaches to building and deploying smart contracts within its ecosystem. As a multi-chain network designed for interoperability, Polkadot provides various environments optimized for different developer preferences and application requirements. From native smart contract support on Asset Hub to specialized parachain environments, developers can choose the platform that best suits their technical needs while benefiting from Polkadot's shared security model and cross-chain messaging capabilities.

> Whether you're looking for Ethereum compatibility through EVM-based parachains like Moonbeam, Astar, and Acala, or prefer WebAssembly-based development with ink!, the Polkadot ecosystem accommodates diverse developer.

Reviewer 2: ```suggestion
Whether you're looking for Ethereum compatibility through EVM-based parachains like Moonbeam, Astar, and Acala or prefer WebAssembly-based development with ink!, the Polkadot ecosystem accommodates diverse developers.
```

> These guides explores the diverse smart contract options available in the Polkadot ecosystem, helping developers understand the unique advantages of each approach and make informed decisions about where to deploy their decentralized applications.

Reviewer 2: ```suggestion
These guides explore the diverse smart contract options available in the Polkadot ecosystem, helping developers understand the unique advantages of each approach and make informed decisions about where to deploy their decentralized applications.
```

- [**Astar**](https://astar.network/){target=\_blank} - Astar distinguishes itself with a unique multiple virtual machine approach supporting both EVM and WebAssembly (Wasm) smart contracts. This dual VM support allows developers to choose their preferred programming environment while maintaining full Ethereum compatibility. Astar has established itself as an innovation hub through initiatives like zk-rollup development and integration with Layer 2 scaling solutions.

- [**Acala**](https://acala.network/){taget=\_blank} - this parachian positions itself as Polkadot's DeFi hub with its Acala EVM+, an enhanced version of the EVM optimized for DeFi operations. This customized implementation enables seamless deployment of Ethereum-based DeFi protocols while offering advanced features like on-chain scheduling, pre-built DeFi primitives, and native multi-token support that aren't available in traditional EVMs.

> All these parachains support familiar Ethereum development tools:

Reviewer 2: As a suggestion, we can add internal links to pages with content we have already created and links to official sites for those without.

- Governance systems integrated with Polkadot's democracy mechanisms
- Cross-chain bridges and asset management solutions

> ## Other Smart Contract Environments

Reviewer: What are the other smart contract environments? How can I quickly understand without having to read everything under this H2?

Author: agreed! fixed in [216f368](https://github.com/polkadot-developers/polkadot-docs/pull/436/commits/216f368f6c0584f74d46a4c003d1b5656376fb84)

### Parachain Contracts

Polkadot's ecosystem includes several parachains that offer EVM compatibility, allowing developers to leverage Ethereum's extensive tooling and developer community while benefiting from Polkadot's security and interoperability features.

> #### Why Adopt the EVM as a Polkadot Parachain?

Reviewer: This seems out of place to me 🤷‍♀️ - who are we targeting with this page?

EVM compatibility provides significant advantages:

- Access to Ethereum's large developer ecosystem and mindshare
- Smart contract portability across chains
- Extensive tooling including wallets, explorers, and development frameworks
- Established security auditors and institutional asset management systems

#### Major EVM-Compatible Parachains

Some of the EVM-compatible parachains are:

> - **Moonbeam** - Moonbeam was the first parachain to bring full Ethereum-compatibility to Polkadot. Its runtime is built using FRAME and combines components from the Polkadot-SDK, Frontier, and custom pallets. Moonbeam serves as an interoperability hub with connections to multiple external chains through various general message passing (GMP) providers like Wormhole, LayerZero, and Axelar.
> 
> - **Astar** - Astar distinguishes itself with a unique multiple virtual machine approach supporting both EVM and WebAssembly (Wasm) smart contracts. This dual VM support allows developers to choose their preferred programming environment while maintaining full Ethereum compatibility. Astar has established itself as an innovation hub through initiatives like zk-rollup development and integration with Layer 2 scaling solutions.
> 
> - **Acala** - this parachian positions itself as Polkadot's DeFi hub with its Acala EVM+, an enhanced version of the EVM optimized for DeFi operations. This customized implementation enables seamless deployment of Ethereum-based DeFi protocols while offering advanced features like on-chain scheduling, pre-built DeFi primitives, and native multi-token support that aren't available in traditional EVMs.

Reviewer: ```suggestion
- **Moonbeam** - Moonbeam was the first parachain to bring full Ethereum-compatibility to Polkadot. Its runtime is built using FRAME and combines components from the Polkadot SDK, Frontier, and custom pallets. Moonbeam serves as an interoperability hub with connections to multiple external chains through various general message passing (GMP) providers like Wormhole, LayerZero, and Axelar

- **Astar** - Astar distinguishes itself with a unique multiple virtual machine approach supporting both EVM and WebAssembly (Wasm) smart contracts. This dual VM support allows developers to choose their preferred programming environment while maintaining full Ethereum compatibility. Astar has established itself as an innovation hub through initiatives like zk-rollup development and integration with Layer 2 scaling solutions

- **Acala** - this parachian positions itself as Polkadot's DeFi hub with its Acala EVM+, an enhanced version of the EVM optimized for DeFi operations. This customized implementation enables seamless deployment of Ethereum-based DeFi protocols while offering advanced features like on-chain scheduling, pre-built DeFi primitives, and native multi-token support that aren't available in traditional EVMs
```

#### Developer Tools Support

All these parachains support familiar Ethereum development tools:

- Hardhat, Remix, Foundry
- Thirdweb
- OpenZeppelin contracts
- Solidity as the primary language

### Wasm (ink!)

> The [pallet_contracts](https://docs.rs/pallet-contracts/latest/pallet_contracts/index.html#contracts-pallet){target=\_blank} is a specialized pallet within the Polkadot SDK that enables smart contract functionality through a WebAssembly (Wasm) execution environment. For developing smart contracts for this pallet, [ink!](https://use.ink/){target=\_blank} emerges as the primary and recommended language.

Reviewer: ```suggestion
The [`pallet_contracts`](https://docs.rs/pallet-contracts/latest/pallet_contracts/index.html#contracts-pallet){target=\_blank} is a specialized pallet within the Polkadot SDK that enables smart contract functionality through a WebAssembly (Wasm) execution environment. For developing smart contracts for this pallet, [ink!](https://use.ink/){target=\_blank} emerges as the primary and recommended language.
```

- Governance systems integrated with Polkadot's democracy mechanisms
- Cross-chain bridges and asset management solutions

## Other Smart Contract Environments

While Asset Hub provides native smart contract support through PolkaVM, Polkadot's ecosystem offers several alternatives for smart contract development. These include EVM-compatible parachains like Moonbeam, Astar, and Acala that support Ethereum tooling and development workflows, as well as WebAssembly-based environments using ink!. Each environment offers unique advantages depending on your development preferences and application requirements.

### Parachain Contracts

Polkadot's ecosystem includes several parachains that offer EVM compatibility, allowing developers to leverage Ethereum's extensive tooling and developer community while benefiting from Polkadot's security and interoperability features.

EVM compatibility provides significant advantages:

- Access to Ethereum's large developer ecosystem and mindshare
- Smart contract portability across chains

> - Extensive tooling including wallets, explorers, and development frameworks

Reviewer 2: ```suggestion
- Extensive tooling, including wallets, explorers, and development frameworks
```

- Established security auditors and institutional asset management systems

Some of the EVM-compatible parachains are:

> - [**Moonbeam**](https://moonbeam.network/){target=\_blank} - Moonbeam was the first parachain to bring full Ethereum-compatibility to Polkadot. Its runtime is built using FRAME and combines components from the Polkadot-SDK, Frontier, and custom pallets. Moonbeam serves as an interoperability hub with connections to multiple external chains through various general message passing (GMP) providers like Wormhole, LayerZero, and Axelar.

Reviewer 2: ```suggestion
- [**Moonbeam**](https://moonbeam.network/){target=\_blank} - Moonbeam was the first parachain to bring full Ethereum-compatibility to Polkadot. Its runtime is built using FRAME and combines components from the Polkadot-SDK, Frontier, and custom pallets. Moonbeam serves as an interoperability hub with connections to multiple external chains through various general message passing (GMP) providers like Wormhole, LayerZero, and Axelar
```

> - [**Astar**](https://astar.network/){target=\_blank} - Astar distinguishes itself with a unique multiple virtual machine approach supporting both EVM and WebAssembly (Wasm) smart contracts. This dual VM support allows developers to choose their preferred programming environment while maintaining full Ethereum compatibility. Astar has established itself as an innovation hub through initiatives like zk-rollup development and integration with Layer 2 scaling solutions.

Reviewer 2: ```suggestion
- [**Astar**](https://astar.network/){target=\_blank} - Astar distinguishes itself with a unique multiple virtual machine approach supporting both EVM and WebAssembly (Wasm) smart contracts. This dual VM support allows developers to choose their preferred programming environment while maintaining full Ethereum compatibility. Astar has established itself as an innovation hub through initiatives like zk-rollup development and integration with Layer 2 scaling solutions
```

> - [**Acala**](https://acala.network/){taget=\_blank} - this parachian positions itself as Polkadot's DeFi hub with its Acala EVM+, an enhanced version of the EVM optimized for DeFi operations. This customized implementation enables seamless deployment of Ethereum-based DeFi protocols while offering advanced features like on-chain scheduling, pre-built DeFi primitives, and native multi-token support that aren't available in traditional EVMs.

Reviewer 2: ```suggestion
- [**Acala**](https://acala.network/){taget=\_blank} - this parachian positions itself as Polkadot's DeFi hub with its Acala EVM+, an enhanced version of the EVM optimized for DeFi operations. This customized implementation enables seamless deployment of Ethereum-based DeFi protocols while offering advanced features like on-chain scheduling, pre-built DeFi primitives, and native multi-token support unavailable in traditional EVMs
```
